features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/aes/aes_alt.c

```diff
@@ -0,0 +1,584 @@
+};
+
+extern volatile int  g_AES_done;
+
+// Must be a multiple of 16 bytes block size 
+#define MAX_DMA_CHAIN_SIZE (16*6)
+MBED_ALIGN(4) static uint8_t au8OutputData[MAX_DMA_CHAIN_SIZE];
+MBED_ALIGN(4) static uint8_t au8InputData[MAX_DMA_CHAIN_SIZE];
+
+static void dumpHex(const unsigned char au8Data[], int len)
+{
+		int j;									
+		for (j = 0; j < len; j++) mbedtls_trace("%02x ", au8Data[j]);									
+		mbedtls_trace("\r\n");										
+}	
+
+static void swapInitVector(unsigned char iv[16])
+{
+	  unsigned int* piv;
+	  int i;
+		// iv SWAP
+		piv = (unsigned int*)iv;
+		for( i=0; i< 4; i++)
+		{
+				*piv = (((*piv) & 0x000000FF) << 24) |
+				(((*piv) & 0x0000FF00) << 8) |
+				(((*piv) & 0x00FF0000) >> 8) |
+				(((*piv) & 0xFF000000) >> 24);
+				piv++;
+		}			
+}	
+
+/* IRQHandler: To share CRYPTO_IRQHandler() with TRNG & other crypto IPs 
+               For ex: 
+                    volatile void CRYPTO_IRQHandler()
+                    {
+                        ...
+                        if (AES_GET_INT_FLAG()) {
+                            g_AES_done = 1;
+                            AES_CLR_INT_FLAG();
+                        }
+                        ...
+                    }               
+*/
+
+
+/* AES available channel 0~3 */
+static unsigned char channel_flag[4]={0x00,0x00,0x00,0x00};  // 0: idle, 1: busy
+static int channel_alloc()
+{
+	int i;
+	for(i=0; i< (int)sizeof(channel_flag); i++)
+	{
+		if( channel_flag[i] == 0x00 )
+		{
+			channel_flag[i] = 0x01;
+			return i;
+		}	
+	}
+	return(-1);
+}
+
+static void channel_free(int i)
+{
+	if( i >=0 && i < (int)sizeof(channel_flag) )
+        channel_flag[i] = 0x00;
+}
+
+
+void mbedtls_aes_init( mbedtls_aes_context *ctx )
+{
+	int i =-1;
+
+	
+    mbedtls_trace("=== %s \r\n", __FUNCTION__);
+    memset( ctx, 0, sizeof( mbedtls_aes_context ) );
+	
+    ctx->swapType = AES_IN_OUT_SWAP;
+    while( (i = channel_alloc()) < 0 ) 	
+    {	
+        mbed_assert_internal("No available AES channel", __FILE__, __LINE__);
+    }
+    ctx->channel = i;
+    ctx->iv = au32MyAESIV;
+	
+    /* Unlock protected registers */
+    SYS_UnlockReg();
+    CLK_EnableModuleClock(CRPT_MODULE);
+    /* Lock protected registers */
+    SYS_LockReg();
+
+    NVIC_EnableIRQ(CRPT_IRQn);
+    AES_ENABLE_INT();    	
+	mbedtls_trace("=== %s channel[%d]\r\n", __FUNCTION__, (int)ctx->channel);
+}
+
+void mbedtls_aes_free( mbedtls_aes_context *ctx )
+{
+
+    mbedtls_trace("=== %s channel[%d]\r\n", __FUNCTION__,(int)ctx->channel);	
+
+    if( ctx == NULL )
+        return;
+      	
+    channel_free(ctx->channel);
+    mbedtls_zeroize( ctx, sizeof( mbedtls_aes_context ) );
+}
+
+/*
+ * AES key schedule (encryption)
+ */
+#if defined(MBEDTLS_AES_SETKEY_ENC_ALT)
+int mbedtls_aes_setkey_enc( mbedtls_aes_context *ctx, const unsigned char *key,
+                    unsigned int keybits )
+{
+    unsigned int i;
+	
+	mbedtls_trace("=== %s keybits[%d]\r\n", __FUNCTION__, keybits);
```

Comment:
@RonEld Please check fix in another PR https://github.com/ARMmbed/mbed-os/pull/4925/commits/b931d7512ea2fa4401a87f603f9a8bf5ca5744e0.